mobile_verifier/src/verified_subscriber_mapping_event.rs

```diff
@@ -0,0 +1,146 @@
+}
+
+impl VerifiedSubscriberMappingEventDeamon {
+    pub fn new(
+        pool: Pool<Postgres>,
+        events: Receiver<FileInfoStream<VerifiedSubscriberMappingEventIngestReport>>,
+    ) -> Self {
+        Self { pool, events }
+    }
+
+    pub async fn create_managed_task(
+        pool: Pool<Postgres>,
+        settings: &Settings,
+        file_store: FileStore,
+    ) -> anyhow::Result<impl ManagedTask> {
+        let (events, event_server) =
+            file_source::continuous_source::<VerifiedSubscriberMappingEventIngestReport, _>()
+                .state(pool.clone())
+                .store(file_store)
+                .lookback(LookbackBehavior::StartAfter(settings.start_after))
+                .prefix(FileType::VerifiedSubscriberMappingEventIngestReportV1.to_string())
+                .create()
+                .await?;
+
+        let task = Self::new(pool, events);
+
+        Ok(TaskManager::builder()
+            .add_task(event_server)
+            .add_task(task)
+            .build())
+    }
+
+    pub async fn run(mut self, shutdown: triggered::Listener) -> anyhow::Result<()> {
+        tracing::info!("Starting sme deamon");
+        loop {
+            tokio::select! {
+                biased;
+                _ = shutdown.clone() => {
+                    tracing::info!("sme deamon shutting down");
+                    break;
+                }
+                Some(file) = self.events.recv() => {
+                    self.process_file(file).await?;
+                }
+            }
+        }
+        Ok(())
+    }
+
+    async fn process_file(
+        &self,
+        file: FileInfoStream<VerifiedSubscriberMappingEventIngestReport>,
+    ) -> anyhow::Result<()> {
+        tracing::info!(
+            "Processing Verified Mapping Event file {}",
+            file.file_info.key
+        );
+        let mut transaction = self.pool.begin().await?;
+        let mut reports = file.into_stream(&mut transaction).await?;
+
+        while let Some(report) = reports.next().await {
+            save_event(&report.report, &mut transaction).await?;
+        }
+
+        transaction.commit().await?;
+        Ok(())
+    }
+}
+
+impl ManagedTask for VerifiedSubscriberMappingEventDeamon {
+    fn start_task(
+        self: Box<Self>,
+        shutdown: triggered::Listener,
+    ) -> futures::future::LocalBoxFuture<'static, anyhow::Result<()>> {
+        Box::pin(self.run(shutdown))
+    }
+}
+
+async fn save_event(
+    event: &VerifiedSubscriberMappingEvent,
+    exec: &mut Transaction<'_, Postgres>,
+) -> Result<(), sqlx::Error> {
+    sqlx::query(
+        r#"
+        insert into verified_mapping_event (subscriber_id, total_reward_points, timestamp)
+        values ($1, $2, $3)
+        on conflict (subscriber_id, timestamp) do nothing
+        "#,
+    )
+    .bind(&event.subscriber_id)
+    .bind(event.total_reward_points as i64)
+    .bind(event.timestamp)
```

Comment:
shouldn't this be the `received_timestamp` of the ingest report?

Reply:
done